OBSDOCS-1775: Cumulative-to-Delta Processor as a sidecar (#94218)
openshift/openshift-docs: observability/otel/otel-collector/otel-collector-processors.adoc

[source,yaml]
----
# ...
mode: sidecar # <1>

> config: |

Author: @pavolloffay, I fixed this. I also checked all other OTel files, and no need to update it elsewhere. However, I opened https://issues.redhat.com/browse/OBSDOCS-1995 .

  processors:
    cumulativetodelta:
      include: # <2>
        match_type: strict # <3>
        metrics: # <4>
        - <metric_1_name>
        - <metric_2_name>
      exclude: # <5>
        match_type: regexp
        metrics:
        - "<regular_expression_for_metric_names>"
# ...
----

> <1> This processor requires a stateful Collector deployment. So you can deploy it as a sidecar to the application that emits the delta temporality metrics.

Reviewer: I am having a hard time understanding this callout. 

Are you saying:
If you want to deploy this processor as a sidecar, the processor requires a stateful collector deployment. You must deploy the sidecar to the application that emits the cumulative temporality metrics.

OR, are you saying:
This processor requires a stateful collector deployment and it must be deployed as a sidecar. You must deploy the sidecar to the application that emits the cumulative temporality metrics.

I also think you might be trying to say something different. TLDR: I think this needs clarification.

> <1> This processor requires a stateful Collector deployment. So you can deploy it as a sidecar to the application that emits the delta temporality metrics.

Author: Thank you. Fixed.

> <1> This processor requires a stateful Collector deployment, which is best achieved by deploying it as a sidecar to the application that emits the delta temporality metrics.

Author: Awesome. I  rewrote it like this already, just started the second sentence with a `So you can ...` to make it a bit more readable.

> <2> Optional: Configures which metrics to include. When omitted, all metrics, except for those listed in the `exclude` field, are converted to delta metrics.

Reviewer 2: ```suggestion
<2> Optional: Configures which metrics to include. If omitted, the processor converts all metrics, except those listed in the `exclude` field, to delta metrics.
```
https://www.ibm.com/docs/en/ibm-style?topic=grammar-conjunctions - suggestion to use if to express a condition.

> <2> Optional: Configures which metrics to include. When omitted, all metrics, except for those listed in the `exclude` field, are converted to delta metrics.

Author: ```suggestion
<2> Optional. You can limit which metrics the processor converts by explicitly defining them here. If you omit this field, the processor converts all metrics, except those listed in the `exclude` field.
```

> <2> Optional: Configures which metrics to include. When omitted, all metrics, except for those listed in the `exclude` field, are converted to delta metrics.

Author: Thank you for the suggestion. This is how I rewrote this callout:
```suggestion
<2> Optional. You can limit which metrics the processor converts by explicitly defining them here. If you omit this field, the processor converts all metrics, except the metrics that are listed in the `exclude` field.
```

> <2> Optional. You can limit which metrics the processor converts. You can do this by explicitly defining these metrics here. If you omit this field, the processor converts all metrics, except the metrics that are listed in the `exclude` field.

Reviewer: ```suggestion
<2> Optional: You can limit which metrics the processor converts by explicitly defining which metrics you want converted in this stanza. If you omit this field, the processor converts all metrics, except the metrics that are listed in the `exclude` field.
```

> <3> Defines a value that you provided in the `metrics` field as a `strict` exact match or `regexp` regular expression.

Reviewer: ```suggestion
<3> Defines the value that you provided in the `metrics` field as an exact match by using the `strict` parameter or a regular expression by using the `regex` parameter.
```

Author: Thank you so much. Applying.

> <4> Lists the metric names, which are exact matches or matches for regular expressions, of the metrics to be converted to delta metrics. If a metric matches both the `include` and `exclude` filters, the `exclude` filter takes precedence.

Reviewer 3: https://www.ibm.com/docs/en/ibm-style?topic=medium-global-audiences#grammar

> <4> Lists the metric names, which are exact matches or matches for regular expressions, of the metrics to be converted to delta metrics. If a metric matches both the `include` and `exclude` filters, the `exclude` filter takes precedence.

Author: Thank you for the suggestion. However, the nonrestrictive clause is here on purpose.
I rewrote the callout, moving the nonrestrictive clause into a new, separate sentence:
```suggestion
<4> Lists the names of the metrics that you want to convert. The processor converts exact matches or matches for regular expressions. If a metric matches both the `include` and `exclude` filters, the `exclude` filter takes precedence.
```

> <5> Optional: Configures which metrics to exclude. When omitted, no metrics are excluded from conversion to delta metrics.

Reviewer 3: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice
